lib/kubernetes-deploy/runner_task.rb

```diff
@@ -8,51 +8,99 @@ module KubernetesDeploy
+    def initialize(namespace:, context:, logger:, max_watch_seconds: nil)
       @logger = logger
       @namespace = namespace
-      @kubeclient = build_v1_kubeclient(context)
       @context = context
+      @max_watch_seconds = max_watch_seconds
     end
 
     def run(*args)
       run!(*args)
       true
-    rescue FatalDeploymentError => error
-      @logger.fatal "#{error.class}: #{error.message}"
+    rescue DeploymentTimeoutError, FatalDeploymentError
       false
     end
 
-    def run!(task_template:, entrypoint:, args:, env_vars: [])
+    def run!(task_template:, entrypoint:, args:, env_vars: [], verify_result: true)
       @logger.reset
-      @logger.phase_heading("Validating configuration")
+
+      @logger.phase_heading("Initializing task")
       validate_configuration(task_template, args)
-      if kubectl.server_version < Gem::Version.new(MIN_KUBE_VERSION)
-        @logger.warn(KubernetesDeploy::Errors.server_version_warning(kubectl.server_version))
+      pod = build_pod(task_template, entrypoint, args, env_vars, verify_result)
+      validate_pod(pod)
+
+      @logger.phase_heading("Running pod")
+      create_pod(pod)
+
+      if verify_result
+        @logger.phase_heading("Streaming logs")
+        watch_pod(pod)
+      else
+        record_status_once(pod)
       end
-      @logger.phase_heading("Fetching task template")
-      raw_template = get_template(task_template)
+      @logger.print_summary(:success)
+    rescue DeploymentTimeoutError
+      @logger.print_summary(:timed_out)
+      raise
+    rescue FatalDeploymentError
+      @logger.print_summary(:failure)
+      raise
+    end
 
-      @logger.phase_heading("Constructing final pod specification")
-      rendered_template = build_pod_template(raw_template, entrypoint, args, env_vars)
+    private
 
-      validate_pod_spec(rendered_template)
+    def create_pod(pod)
+      @logger.info "Creating pod '#{pod.name}'"
+      pod.deploy_started_at = Time.now.utc
+      kubeclient.create_pod(pod.to_kubeclient_resource)
+      @pod_name = pod.name
+      @logger.info("Pod creation succeeded")
+    rescue KubeException => e
+      msg = "Failed to create pod: #{e.class.name}: #{e.message}"
+      @logger.summary.add_paragraph(msg)
+      raise FatalDeploymentError, msg
+    end
 
-      @logger.phase_heading("Creating pod")
-      @logger.info("Starting task runner pod: '#{rendered_template.metadata.name}'")
-      @kubeclient.create_pod(rendered_template)
+    def build_pod(template_name, entrypoint, args, env_vars, verify_result)
+      task_template = get_template(template_name)
+      @logger.info("Using template '#{template_name}'")
+      pod_template = build_pod_definition(task_template)
+      set_container_overrides!(pod_template, entrypoint, args, env_vars)
+      ensure_valid_restart_policy!(pod_template, verify_result)
+      Pod.new(namespace: @namespace, context: @context, logger: @logger, stream_logs: true,
```

Comment:
This option is really about _how_ the logs are displayed when the pod is being watched, not _if_: "stream logs as we get them" vs. "dump them all when this succeeds/fails". If you don't start watching this pod at all (because `verify_result` is false), `stream_logs` is irrelevant. Currently `stream_logs` is always true in task runs and always false in deploys. As discussed with Danny in a comment above, I think it would be cool if we could intelligently set it to true during certain deploys too.